modules/react/toast/lib/Toast.tsx

```diff
@@ -1,118 +1,83 @@
+ *      <Toast.Icon icon={checkIcon} color={colors.greenApple400} />
+ *      <Toast.Body>
+ *        <Toast.Message>Your workbook was successfully processed.</Toast.Message>
+ *        <Toast.Link href="#hreflink">Custom Link</Toast.Link>
+ *      </Toast.Body>
+ *      <Toast.CloseIcon aria-label="Close" onClick={handleClose} />
+ *    </Toast>
+ * );
+ *```
+ */
+export const Toast = createContainer('div')({
   displayName: 'Toast',
-  Component: (
-    {
-      icon = checkIcon, // needed for TS2742 - https://github.com/microsoft/TypeScript/issues/29808
-      iconColor = colors.greenApple400,
-      onClose,
-      onActionClick,
-      actionText,
-      children,
-      ...elemProps
-    }: ToastProps,
-    ref,
-    Element
-  ) => {
-    const isInteractive = onClose || onActionClick;
-    const isError = iconColor === colors.cinnamon500 && icon === exclamationCircleIcon;
-
-    return (
-      <Popup.Card
-        ref={ref}
-        as={Element}
-        width={toastWidth}
-        depth={5}
-        role={isInteractive ? 'dialog' : isError ? 'alert' : 'status'}
-        aria-live={isInteractive ? 'off' : isError ? 'assertive' : 'polite'}
-        aria-atomic={!isInteractive}
-        padding="xxs"
-        {...elemProps}
-      >
-        {onClose && <Popup.CloseIcon aria-label="Close" onClick={onClose} size="small" />}
-        <Popup.Body>
-          <ToastContentContainer onClose={onClose}>
-            {icon && <ToastSystemIcon color={iconColor} colorHover={iconColor} icon={icon} />}
-            <Message>
-              {children}
-              {onActionClick && (
-                <StyledActionButton as="button" onClick={onActionClick}>
-                  {actionText}
-                </StyledActionButton>
-              )}
-            </Message>
-          </ToastContentContainer>
-        </Popup.Body>
-      </Popup.Card>
-    );
+  modelHook: useToastModel,
+  subComponents: {
+    Body: ToastBody,
+    CloseIcon: ToastCloseIcon,
+    Icon: ToastIcon,
+    Message: ToastMessage,
+    Link: ToastLink,
   },
+})<ToastProps>(({children, ...elemProps}, Element, model) => {
+  const getAriaAttributes = (mode: string): React.HtmlHTMLAttributes<HTMLDivElement> => {
+    switch (mode) {
+      case 'dialog':
+        return {
+          'aria-describedby': model.state.id,
+          'aria-labelledby': undefined,
+          role: 'dialog',
+        };
+      case 'alert':
+        return {
+          role: 'alert',
+          'aria-live': 'assertive',
+          'aria-atomic': true,
+        };
+      case 'status':
+        return {
+          role: 'status',
+          'aria-live': 'polite',
+          'aria-atomic': true,
+        };
+      default: {
+        return {};
+      }
+    }
+  };
+  return (
+    <Popup.Card
+      as={Flex}
+      width={360}
+      padding="0"
+      {...getAriaAttributes(model.state.mode)}
+      flexDirection="row"
+      gap="xxxs"
+      {...elemProps}
+    >
+      {children}
```

Comment:
There use to be a `flex` component wrapped around children, i changed this to clean up the dom